pkg/kubectl/cmd/cmd.go

```diff
@@ -109,7 +109,12 @@ __kubectl_parse_get()
 __kubectl_get_resource()
 {
     if [[ ${#nouns[@]} -eq 0 ]]; then
-        return 1
+      local kubectl_out
+      if kubectl_out=$(kubectl api-resources -o name --cached --request-timeout=5s --verbs=get 2>/dev/null); then
```

Comment:
In order to make it possible to override `--context` flag, you need to add `$(__kubectl_override_flags)` to arguments of `kubectl api-resources`.

```
 if kubectl_out=$(kubectl api-resources $(__kubectl_override_flags) -o name --cached --request-timeout=5s --verbs=get 2>/dev/null); then
```

Reply:
thanks, updated. also, fixed a bug in `__kubectl_override_flags()` that was making use of `--all-namespaces` (which isn't supported for any of the places `__kubectl_override_flags` was being used (config and api-resources don't support that flag, get containers and get resource cannot get by name when combined with --all-namespaces.